docs/DOCUMENTATION.md

```diff
@@ -0,0 +1,872 @@
+  f. The last aggregated price on chain
+  g. The request id of the last result received from the CL node
+  h. The details of the latest on-chain round including the round id, the timestamp when it was started, by whom it was started and whether the oracle made a submission for that round
+
+#### Monitoring
+
+The monitoring script is needed to:
+1. Monitor the actual price and that it is being updated
+2. Monitor node operators submissions to ensure they are submitting values within accepatable thresholds and that they are submitting to rounds and not missing them
+3. Monitor node operators' balances to ensure they have enough balance for transaction fees to push prices on chain
+4. Monitor the rate at which rounds are being created
+
+The monitoring script should contain the following functionalities:
+
+1. A state file which is updated so that whenever the monitoring script is restarted it is able to continue from where it was stopped.
+2. Monitor multiple oracles at once
+3. An endpoint to expose prometheus metrics 
+4. Expose the following metrics
+  a. The latest submitted value on chain by an oracle for a feed
+  b. The timestamp in which an oracle made an on-chain submission for a feed 
+  c. The last round for which an oracle made an on-chain submission for a feed
+  d. The deviation of an oracle's submitted price from the latest aggregated value on-chain for a feed
+  e. The oracle balance 
+  f. The actual price on-chain
+4. Obtain an oracle's invitation IDs for feeds from wallets
+5. Query the latest prices and round submissions of oracles every X seconds and update the metrics
+6. Have an efficient way of polling only the latest price pushes so be able to monitor oracles efficiently when the number of offers used start to increase
+
+
+## File structure
+
+In this section, one is able to find a description of the file structure and the contents of files
+
+#### scripts
+
+The scripts directory contains scripts which are used to for deployments. 
+
+The following are the different scripts which can be found in this directory
+
+1. accept-oracle-invitation.sh - This script can be used to accept an oracle invitation. This script takes in 3 parameters, the wallet name, the brand in and brand out. An example of a command to run this is ```./accept-oracle-invitation.sh $WALLET_NAME $BRAND_IN $BRAND_OUT```
+2. get-sdk-package-names.sh - This script was copied from the agoric-sdk repository and it is used to get the sdk package names.
+3. npm-audit-fix.sh - This script was copied from the agoric-sdk repository and it is used to fix issues with npm.
+4. provision-wallet.sh - This script can be used to provision a smart wallet. This script takes in one parameter, the wallet name. An example of a command to run this is ```./provision-wallet.sh $WALLET_NAME```
+
+#### src
+
+This directory includes all the source code. It is split into two other directories, <b>oracle</b> and <b>lib</b>.
```

Comment:
explain the role for each folder

Reply:
This is done in the folders' respective subsections